packages/dashboard/src/pages/CreateMaskWallet/Recovery/index.tsx

```diff
@@ -131,8 +144,14 @@ const Recovery = memo(function Recovery() {
                 )
                 await WalletServiceRef.value.resolveMaskAccount([{ address }])
                 navigate(DashboardRoutes.SignUpMaskWalletOnboarding, { replace: true })
-            } catch {
-                setError(t.create_wallet_key_store_incorrect_password())
+            } catch (error) {
+                const errorMsg = (error as Error).message
+                // Todo: SDK should return 'Incorrect Keystore Password.' when keystore pwd is wrong.
```

Comment:
Did you mean here?

```ts
export async function recoverWalletFromKeyStoreJSON(name: string, json: string, jsonPassword: string) {
    const password_ = await password.INTERNAL_getPasswordRequired()
    const imported = await Mask.importJSON({
        coin: api.Coin.Ethereum,
        json,
        keyStoreJsonPassword: jsonPassword,
        name,
        password: password_,
    })

    // failed to restore keystore JSON file
    if (!imported?.StoredKey) throw new Error('Failed to import the wallet.')
    const created = await Mask.createAccountOfCoinAtPath({
        coin: api.Coin.Ethereum,
        derivationPath: null,
        name,
        password: password_,
        StoredKeyData: imported.StoredKey.data,
    })
    if (!created?.account?.address) throw new Error('Failed to create the wallet.')
    return database.addWallet(ImportSource.UserProvided, created.account.address, {
        name,
        storedKeyInfo: imported.StoredKey,
    })
}
```

Reply:
No, the definition of error message is not in this codebase.